azure_media_services/static/js/studio_edit.js

```diff
@@ -193,6 +195,113 @@ function StudioEditableXBlockMixin(runtime, element) {
+        $containerCaptions.find('[name = "captions"]:checked').each(function() {
+            var caption = {
+                kind: 'subtitles',
+                src: $(this).val(),
+                srclang: $(this).data('srclang'),
+                label: $(this).data('label')
+            };
+            captions.push(caption);
+        });
+        $(element).find('[data-field-name = "captions"] textarea').val(JSON.stringify(captions))
+            .trigger('change');
+    }
+
+    /**
+     * setOnChangeCaptions
+     */
+    function setOnChangeCaptions() {
+        $containerCaptions.find('[name = "captions"]').on('change', setCaptionsField);
+    }
+
+    /**
+     * renderCaptions
+     * @param data
+     */
+    function renderCaptions(data) {
+        var i, html;
+        $containerCaptions.empty();
+        if (data.length === 0) {
+            $containerCaptions.text(gettext('No captions/transcripts available for selected video.'));
+        } else {
+            for (i = 0; i < data.length; i++) {
+                html = '<li class="select-holder"><div class="wrap-input-captions"><input id="checkbox-captions-' + i +
+                        '" type="checkbox" name="captions" value="' + data[i].download_url +
+                        '" data-srclang="' + data[i].language + '" data-label="' + data[i].language_title +
+                        '"/><label for="checkbox-captions-' + i + '">' + data[i].file_name +
+                        ' (' + data[i].language + ')</label></div></li>';
+                $containerCaptions.append(html);
+            }
+            setOnChangeCaptions();
+        }
+    }
+
+    /**
+     * setVideoInfo
+     * @param videInfo
+     */
+    function setVideoInfo(videInfo) {
+        $(element).find('[data-field-name = "download_url"] input').val(videInfo.download_video_url)
+            .trigger('change');
+        $(element).find('[data-field-name = "video_url"] input').val(videInfo.smooth_streaming_url)
+            .trigger('change');
+    }
+
+    /**
+     * resetCaptionsField
+     */
+    function resetCaptionsField() {
+        $(element).find('[data-field-name = "captions"] textarea').val(JSON.stringify([]))
+            .trigger('change');
+    }
+
+    /**
+     * getCaptionsAndVideoInfo
+     * @param edxVideoID
+     */
+    function getCaptionsAndVideoInfo(edxVideoID) {
+        $containerCaptions.html('<div class="loader-wrapper"><span class="loader"><svg class="icon icon-spinner11">' +
+            '<use xlink:href="#icon-spinner11"></use></svg></span></div class="loader-wrapper">');
+        $.ajax({
+            type: 'POST',
+            url: handlerUrlGetCaptionsAndVideoInfo,
+            data: JSON.stringify({edx_video_id: edxVideoID}),
+            dataType: 'json',
+            success: function(data) {
+                if (data.error_message !== '') {
+                    $containerCaptions.html('<span class="ams-info">' + data.error_message + '</span>');
```

Comment:
We should address all of these cases. Pretty sure you can use _.escape(dynamic_data) to html escape.
Should be done everywhere there is dynamic html